skimage/filters/_gabor.py

```diff
@@ -151,25 +360,32 @@ def gabor(image, frequency, theta=0, bandwidth=1, sigma_x=None,
     >>> from skimage.filters import gabor
     >>> from skimage import data, io
     >>> from matplotlib import pyplot as plt  # doctest: +SKIP
-
     >>> image = data.coins()
     >>> # detecting edges in a coin image
     >>> filt_real, filt_imag = gabor(image, frequency=0.6)
     >>> plt.figure()            # doctest: +SKIP
     >>> io.imshow(filt_real)    # doctest: +SKIP
     >>> io.show()               # doctest: +SKIP
-
     >>> # less sensitivity to finer details with the lower frequency kernel
     >>> filt_real, filt_imag = gabor(image, frequency=0.1)
     >>> plt.figure()            # doctest: +SKIP
     >>> io.imshow(filt_real)    # doctest: +SKIP
     >>> io.show()               # doctest: +SKIP
     """
-    assert_nD(image, 2)
-    g = gabor_kernel(frequency, theta, bandwidth, sigma_x, sigma_y, n_stds,
-                     offset)
+    if kernel is None:
+        if frequency is None:
+            raise TypeError("gabor() must specify 'frequency' "
+                            "if 'kernel' is not provided")
```

Comment:
I think but am not sure that missing arguments should raise a ValueError, not a TypeError.

Reply:
Good point! I thought a missing argument was a ValueError but actually it's a TypeError. I guess it's because the "type" of the function is not being respected. 

Carry on. =P